File: sub-system/src/main/java/org/github/service/IShiroService.java
package org.github.service;

import lombok.NonNull;
import org.github.shiro.AuthorInfo;
import org.github.shiro.User;

import java.util.List;

public interface IShiroService {
    @NonNull List<AuthorInfo> queryAuthorInfo(@NonNull String userId);

    @NonNull User queryUser(@NonNull String username);
}

File: sub-system-config/src/main/java/org/github/service/IShiroService.kt
package org.github.service

import org.github.shiro.AuthorInfo
import org.github.shiro.User
import org.github.spring.restful.json.JSONArrayReturn
import org.github.spring.restful.json.JSONDataReturn

interface IShiroService {
  fun queryAuthorInfo(userId: String): JSONArrayReturn<AuthorInfo>

  fun queryUser(username: String): JSONDataReturn<out User>
}
